_migration-assistant/migration-phases/planning-your-migration/handling-type-mapping-deprecation.md

```diff
@@ -0,0 +1,189 @@
+
+If you have an index `activity` with types `user` and `post` that you want to split into separate indices:
+
+```json
+[
+  {
+    "TypeMappingsSanitizationTransformerProvider": {
+      "staticMappings": {
+        "activity": {
+          "user": "new_users",
+          "post": "new_posts"
+        }
+      },
+      "sourceProperties": {
+        "version": {
+          "major": 6,
+          "minor": 8
+        }
+      }
+    }
+  }
+]
+```
+
+This will:
+- Route documents with type `user` to the `new_users` index
+- Route documents with type `post` to the `new_posts` index
+
+#### 2. Merge All Types into One Index
+
+To combine multiple types into a single index:
+
+```json
+[
+  {
+    "TypeMappingsSanitizationTransformerProvider": {
+      "staticMappings": {
+        "activity": {
+          "user": "combined_activity",
+          "post": "combined_activity"
+        }
+      },
+      "sourceProperties": {
+        "version": {
+          "major": 6,
+          "minor": 8
+        }
+      }
+    }
+  }
+]
+```
+
+#### 3. Drop Specific Types
+
+To migrate only specific types and drop others:
+
+```json
+[
+  {
+    "TypeMappingsSanitizationTransformerProvider": {
+      "staticMappings": {
+        "activity": {
+          "user": "users_only",
+        }
+      },
+      "sourceProperties": {
+        "version": {
+          "major": 6,
+          "minor": 8
+        }
+      }
+    }
+  }
+]
+```
+This configuration will only migrate documents of type `user` and ignore other types in the `activity` index.
+
+### 4. Keep original structure
+
+To migrate only specific types and keep the original structure:
+
+```json
+[
+  {
+    "TypeMappingsSanitizationTransformerProvider": {
+      "regexMappings": [
+        [
+          "(.*)",
+          ".*",
+          "$1"
+        ]
+      ],
+      "sourceProperties": {
+        "version": {
+          "major": 6,
+          "minor": 8
+        }
+      }
+    }
+  }
+]
+```
+
+This is equivalent to the merge all types into one index rule but using a pattern-based routing strategy.
+
+### Defaults
+
+When no regex mappings are specified, the transformer will default to the following behavior:
+
+```json
+{
+  "regexMappings": [
+    ["(.+)", "_doc", "$1"],
+    ["(.+)", "(.+)", "$1_$2"]
+  ],
+}
+```
+
+This has the effect of retaining the index name for es 6+ created indices while comibining the type and index name for es 5.x created indices. If you wish to retain the index name for es 5.x created indices, please use the `staticMappings` option or override the type mappings using the `regexMappings` option similar to the `Keep original structure`
+
+### Limitations
+
+- The transformer cannot be used prior to Elasticsearch 6.x.
+- Within the replayer, the transformer only supports a subset of requests that may be replayed with types, in particular:
+  - PUT/POST /{index}/{type}/{id} 
+  - PUT/POST /{index}/{type}/
+  - GET /{index}/{type}/{id}
+  - PUT/POST /_bulk
+  - PUT/POST /{index}
```

Comment:
Lets turn this into a table with a leading column for the operation name such as `Index`, etc `Bulk Index/Update/Delete`.